core/graphman_extensions/src/handle_broken_executions.rs

```diff
@@ -0,0 +1,145 @@
+use std::future::Future;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use graphman_primitives::BoxedError;
+use graphman_primitives::BoxedFuture;
+use graphman_primitives::ExtensibleGraphmanContext;
+use graphman_primitives::GraphmanCommand;
+use graphman_primitives::GraphmanLayer;
+
+use crate::context_extensions::CommandKind;
+use crate::error::error_builder;
+use crate::GraphmanExtensionError;
+use crate::GraphmanExtensionStore;
+
+error_builder!("HandleBrokenExecutions");
+
+// Command executions will be considered broken if they have
+// not received any updates for this duration.
+const DEFAULT_MAX_INACTIVE_TIME: Duration = Duration::from_secs(300);
+
+/// Marks command executions as failed if they did not receive any
+/// updates for the specified duration.
```

Comment:
There's a setup for running jobs periodically in `store/postgres/src/jobs.rs`. Couldn't we just add a job that periodically goes through executions and marks them as failed?

Reply:
That's a nice idea, but after the refactor I removed that functionality for now because it's not strictly necessary